### DISCUSSION TITLE
EasyPanel - "Bad Gateway" Error.

### DISCUSSION THREAD

**Discussion by amalraj1264** [easypanel-io/community#96] [Q&A]
Hello,

I'm using the free version of Easy Panel on a VPS with Ubuntu Server. I've successfully hosted a WordPress website and a React website through Easy Panel. However, I'm encountering issues when attempting to host MongoDB and Mongo Express. Although they install without any problems, I can't access them through the URL as it returns a 'Bad Gateway' error.

Additionally, I've tried hosting a very basic Django project, which only requires simple configurations, but I face the same issue – it can't be accessed through the URL and also gives a 'bad gateway' error.

Finally, I attempted to deploy Django directly on Ubuntu without using Easypanel, it was successful and running, but I want to migrate it to the Easypanel for easy management.

I want to resolve the "bad gateway" issue. Could it be related to DNS settings on the server, or something else? I don't have in-depth knowledge about these matters and I've tried reaching out to Easy Panel support without success, possibly due to using the free plan.

NB: If I Turn on the maintenance mode of an application (the apps cause "bad gateway") , the maintenance screen of the easypanel through the same URL is working. That means the URL is working but not application.

If anyone has suggestions or a fix, I would greatly appreciate it.

Thank you.

**Comment by deiucanta** [comment]
Hi Amal,

In the domains section, there is an "internal port" that you need to configure. Can you make sure that port is set correctly?

"Bad gateway" means that the connection between Easypanel and your app is broken. Most of the times is because of a bad internal port config.

**Comment by amalraj1264** [comment]
Hi Andrei,

I really appreciate your support.

This domain is auto-generated. I think it's fine to use this domain for
testing purposes. Port 9030 is a free port on my VPS, and I've allowed it
through the firewall as well.

However, I'm still getting a "Bad Gateway" error.

Is there anything more specific I can look for to troubleshoot this issue?


Best,
Amal Rajan

**Comment by deiucanta** [comment]
Send me a screenshot of your Domains page.

**Reply by amalraj1264** [reply]
![Screenshot 2024-06-10 181536](https://github.com/easypanel-io/community/assets/46672898/18866c21-ef48-48d6-b46d-e30231adbb4c)

**Comment by deiucanta** [comment]
I see the yellow indicator on the service. Meaning it's not starting properly. Can you check the deployment logs?

Also, did you know that the Mongo service has Mongo Express built-in? Just go to the Mongo service, and you'll see a section for Mongo Express. Then click enable and open.

**Comment by amalraj1264** [comment]
Yes, at first, I tried to open Mongo Express from the Mongo service, but it was also showing a bad gateway error. That's why I tried a Mongo Express template.

Please check the attached screenshot of the Mongo service, Still it shows "bad gateway" when I try to open Mongo Express by clicking "open" button in "MongoExpress" Section in Mongo Service.

![Screenshot 2024-06-11 113722](https://github.com/easypanel-io/community/assets/46672898/2e40aab6-c600-4d12-8ec1-5ea99dd557d0)
![Screenshot 2024-06-11 113955](https://github.com/easypanel-io/community/assets/46672898/30183f29-e85f-4eac-8d83-66d6fff9ad25)
![Screenshot 2024-06-11 114115](https://github.com/easypanel-io/community/assets/46672898/c85782a0-aee1-49d8-8695-d21fafa90779)

**Comment by amalraj1264** [comment]
I found the issue and resolved it. The problem was that I was using the MongoDB password "Digital@123". MongoDB Compass was connecting to DB with the connection string but, EasyPanel couldn't resolve the hostname with this password automatically. So, I changed the password to one without the "@" symbol, and everything's working fine now.
